File: numconv.ts
namespace numconv {
    //  Add your code here
    export function toHex(dec: number): string {
        let digits = "0123456789ABCDEF"
        let x = dec % 16
        let rest = Math.idiv(dec, 16)
        let s = digits[x]
        while (rest > 0) {
            x = rest % 16
            rest = Math.idiv(rest, 16)
            s = digits[x] + s
        }
        return s
    }
    
    export function toBin(dec: number): string {
        let digits = "01"
        let x = dec % 2
        let rest = Math.idiv(dec, 2)
        let s = digits[x]
        while (rest > 0) {
            x = rest % 2
            rest = Math.idiv(rest, 2)
            s = digits[x] + s
        }
        return s
    }
    
}

File: numconv.py
# Add your code here
def toHex(dec):
    digits = "0123456789ABCDEF"
    x = dec % 16
    rest = dec // 16
    s = digits[x]
    while rest > 0:
        x = rest % 16
        rest = rest // 16
        s = digits[x]+s
    return s

def toBin(dec):
    digits = "01"
    x = dec % 2
    rest = dec // 2
    s = digits[x]
    while rest > 0:
        x = rest % 2
        rest = rest // 2
        s = digits[x]+s
    return s